components/SpotForm.js

```diff
@@ -0,0 +1,172 @@
+import { useState } from "react";
+import { useRouter } from "next/router";
+import styled from "styled-components";
+import { formTaskOptions } from "../helpers/formTaskOptions";
+import { formEquipmentOptions } from "../helpers/formEquipmentOptions";
+import { v4 as uuidv4 } from "uuid";
+
+export default function SpotForm({ addSpots }) {
+  const [formData, setFormData] = useState({
+    id: Date.now(),
+    Spotname: "",
+    Address: "",
+    Tasks: [],
+    Equipment: [],
+  });
+
+  const router = useRouter();
+
+  const handleOptionToggle = (option, field) => {
+    setFormData((prevFormData) => {
+      const selectedOptions = prevFormData[field];
+
+      if (selectedOptions.includes(option)) {
+        return {
+          ...prevFormData,
+          [field]: selectedOptions.filter(
+            (selectedOption) => selectedOption !== option
+          ),
+        };
+      } else {
+        return {
+          ...prevFormData,
+          [field]: [...selectedOptions, option],
+        };
+      }
+    });
+  };
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+
+    setFormData((prevFormData) => ({
+      ...prevFormData,
+      [name]: value,
+    }));
+  };
+
+  const onSubmitHandler = (event) => {
+    event.preventDefault();
+    const newSpot = {
+      id: formData.id,
```

Comment:
This would be the right spot to call the `uuidv4` function to generate the `id`.